EIPS/eip-2124.md

```diff
@@ -0,0 +1,258 @@
+---
+eip: 2124
+title: Zero RTT netsplit on chain mismatch
+author: Péter Szilágyi (peterke@gmail.com)
+discussions-to: https://github.com/ethereum/EIPs/issues/2125
+status: Draft
+type: Standards Track
+category: Networking
+requires: 778
+created: 2019-05-03
+---
+
+## Abstract
+
+There are many public and private Ethereum networks, but the discovery protocol doesn't differentiate between them. The only way to check if a peer is good or bad (same chain or not), is to establish a TCP/IP connection, wrap it with RLPx cryptography, then execute an `eth` handshake. This is an extreme cost to bear if it turns out that the remote peer is on a different network. This cost is magnified for small networks, where a lot more trial and errors are needed to find good nodes.
+
+Even if the peer **is** on the same network, during non-controversial consensus upgrades, not everybody updates their nodes in time (developer nodes, leftovers, etc). These stale nodes put a meaningless burden on the peer-to-peer network, since they just latch on to good nodes, but don't accept upgraded blocks. This causes valuable peer slots and bandwidth to be lost until the stale nodes finally update. This is an even more pronounced issue for test networks, where leftover nodes can linger for many months.
+
+This EIP proposes an enhancement to the ENR extension of the discovery protocol to detect when two nodes are incompatible and never connect them in the first place, instead of wasting resources in vain. The EIP solves a a number of issues:
```

Comment:
Can you make "ENR" a link to https://eips.ethereum.org/EIPS/eip-778. I know you have included the `requires:` header, but it helps the reader to expand on ENR at the first instance it is mentioned.

Reply:
Expanded a bit and added the link, PTAL